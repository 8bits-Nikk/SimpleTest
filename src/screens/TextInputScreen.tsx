import React, {useState} from 'react';
import {SafeAreaView, StyleSheet, TextInput, View} from 'react-native';
import NextnPrevious, {Stack} from '../components/NextnPrevious';

const TextInputScreen = () => {
  const [text, setText] = useState('');
  return (
    <SafeAreaView style={styles.body}>
      <View style={styles.subBody}>
        <TextInput
          placeholder={'Name'}
          testID={'TextInput'}
          value={text}
          onChangeText={txt => setText(txt)}
          style={{
            borderRadius: 12,
            borderWidth: 2,
            borderColor: '#000',
            padding: 16,
            fontSize: 28,
            marginHorizontal: 8,
          }}
        />
      </View>
      <NextnPrevious currentScreen={Stack.TextInputScreen} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
  },
  subBody: {
    flex: 1,
    justifyContent: 'center',
  },
});

export default TextInputScreen;
