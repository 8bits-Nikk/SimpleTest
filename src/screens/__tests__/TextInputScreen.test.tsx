import {renderWithNavigation} from '../../helper/renderWithNavigation';
import TextInputScreen from '../TextInputScreen';
import {fireEvent, waitFor} from '@testing-library/react-native';

describe('TextInputScreen Tests', () => {
  it('should render correctly', () => {
    const tree = renderWithNavigation(<TextInputScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have TextInput', () => {
    const tree = renderWithNavigation(<TextInputScreen />);
    const input = tree.queryByPlaceholderText('Name');
    expect(input).not.toBeNull();
  });

  it('should enter name', async () => {
    const tree = renderWithNavigation(<TextInputScreen />);
    const input = tree.getByTestId('TextInput');
    await waitFor(() => {
      fireEvent.changeText(input, 'Tests');
      expect(input.props.value).toBe('Tests');
    });
  });
});
