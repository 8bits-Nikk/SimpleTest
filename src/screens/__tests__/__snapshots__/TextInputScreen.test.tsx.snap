// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputScreen Tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Name"
      style={
        Object {
          "borderColor": "#000",
          "borderRadius": 12,
          "borderWidth": 2,
          "fontSize": 28,
          "marginHorizontal": 8,
          "padding": 16,
        }
      }
      testID="TextInput"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Previous_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Previous
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Next_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
